examples/graphene_apps/python_worker/avalon_worker/receive_request.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import zmq
+import logging
+import socket
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+# -------------------------------------------------------------------------
+
+
+class ZmqSocket():
+    """
+    ZMQ socker to receive Work Order Request and send Response.
+    """
+
+# -------------------------------------------------------------------------
+
+    def __init__(self, zmq_url, wo_processor):
+        """
+        Constructor for ZmqSocket.
+        """
+        self.wo_processor = wo_processor
+        self.zmq_url = zmq_url
+
+# -------------------------------------------------------------------------
+
+    def start_zmq_listener(self):
+        """
+        This function binds to the port configured for zmq and
+        then indefinitely processes work order requests received
+        over the zmq connection. It terminates only when an
+        exception occurs.
+        """
+        # Binding with ZMQ Port
+        try:
+            socket = self._bind_zmq_socket()
+            logger.info("Bind to zmq port")
+        except Exception as ex:
+            logger.exception("Failed to bind socket" +
+                             "shutting down")
+            logger.error("Exception: {} args{} details{}"
+                         .format(type(ex), ex.args, ex))
+        try:
+            while True:
+                # Wait for the next request
+                logger.info("waiting for next request")
+                msg = socket.recv()
+                msg = msg.decode()
+                logger.info("Received request: {}".format(msg))
+                result = self.wo_processor.process_work_order(msg)
+                logger.info(result)
+                if result:
+                    socket.send_string(result)
+                else:
+                    socket.send_string("Work order result is empty")
+        except Exception as inst:
```

Comment:
Do you intend to exit on any exception? Or are the exceptions that could be suppressed/logged for a request and carry on with the loop. Though I see that in process_work_order() as well you are not raising any exception.